src/mono/mono/mini/method-to-ir.c

```diff
@@ -5663,6 +5663,108 @@ is_addressable_valuetype_load (MonoCompile* cfg, guint8* ip, MonoType* ldtype)
 	return is_load_instruction && is_in_previous_bb && is_struct;
 }
 
+/*
+ * check_get_virtual_method_assumptions:
+ * 
+ * This shadows mono_class_get_virtual_method, but instead of actually resolving
+ * the virtual method, this only checks if mono_class_get_virtual_method would
+ * succeed. This is in place because that function fails catastrophically in some
+ * cases, bringing down the entire runtime. Returns TRUE if the function is safe 
+ * to call, FALSE otherwise.
+ */
+static gboolean
+check_get_virtual_method_assumptions (MonoClass* klass, MonoMethod* method)
```

Comment:
Does it make sense to be in the same file as `mono_class_get_virtual_method`?

Reply:
Sure, it could be moved there. It is currently used only in this one instance, so I made it static and kept it in the same file as the the rest of the mechanism.